File: docusign-sdk-sample-objc/docusign-sdk-sample-objc/Controllers/DocuSignViewController.m
//
//  DocuSignViewController.m
//  docusign-sdk-sample-objc
//
//  Copyright © 2017 DocuSign. All rights reserved.
//


#import "DocuSignViewController.h"

#import "AppDelegate.h"
#import <DocuSignSDK/DocuSignSDK.h>
#import "ProfileManager.h"
#import "TemplatesManager.h"


@interface DocuSignViewController ()
@end


@implementation DocuSignViewController


#pragma mark - UI Lifecycle Methods

- (void)viewDidLoad
{
    [super viewDidLoad];
    [self styleUIElements];

    // set sdk styling
    [DSMAppearance setNavigationBarTintColor:[ProfileManager getAppNavColor]];
    [DSMAppearance setBarButtonItemsTintColor:[ProfileManager getAppBarItemsColor]];

    ProfileManager * profileManager = [ProfileManager sharedInstance];
    // retrieve the selected template id
    NSString * templateId = [profileManager getClientTemplateId];
    
    // retrieve client data for insertion into template
    NSDictionary * tabData = [profileManager getTemplateTabData];
    
    BOOL useOffline = [profileManager useOfflineFlow];
    
    NSString * attachmentPath = [profileManager getAttachmentPath];
    
    // display template via sdk
    [[TemplatesManager sharedInstance] displayTemplateForSignature:templateId forController:self withTabs:tabData useOfflineFlow:useOffline withAttachmentAtPath:attachmentPath withCompletion:^(UIViewController * controller, NSError *error)
     {
         NSLog(@"Done displaying template.");

         // return to claims screen
         [self.navigationController popToRootViewControllerAnimated:TRUE];
     }];
}


- (void) viewWillAppear:(BOOL)animated
{
    // allow landscape orientation for sdk screens
    AppDelegate * appDelegate = (AppDelegate *)[UIApplication sharedApplication].delegate;
    appDelegate.restrictRotation = NO;
}


- (void) styleUIElements
{
    // add custom nav title
    UIView * navTitleView = [[[NSBundle mainBundle] loadNibNamed:@"CustomNavTitle" owner:self options:nil] lastObject];
    self.navigationItem.titleView = navTitleView;
}


@end

File: docusign-sdk-sample-swift/docusign-sdk-sample-swift/Controllers/DocuSignViewController.swift
//
//  DocuSignViewController.swift
//  docusign-sdk-sample-swift
//
//  
//  Copyright © 2017 DocuSign. All rights reserved.
//


import DocuSignSDK
import UIKit


class DocuSignViewController: UIViewController
{

    // MARK: UI Lifecycle Methods

    override func viewDidLoad()
    {
        super.viewDidLoad()
        self.styleUIElements();

        // set sdk styling
        DSMAppearance.setNavigationBarTintColor(UIColor(red: 46.0/255, green: 109.0/255, blue: 164.0/255, alpha: 1.0));
        DSMAppearance.setBarButtonItemsTintColor(UIColor.white);

        let templateId: String = ProfileManager.sharedInstance.getCurrentTemplateId();
        let tabData: Dictionary = ProfileManager.sharedInstance.getTemplateTabData(templateId:templateId);
        let recipientData: Array = ProfileManager.sharedInstance.getTemplateRecipientData(templateId:templateId);
        let onlineSign: Bool = !ProfileManager.sharedInstance.getUseOfflineFlow();
        let attachmentUrl: URL? = ProfileManager.sharedInstance.getAttachmentUrl();
        let customFields = ProfileManager.sharedInstance.getCustomFieldsData(templateId:templateId)

        // display template via sdk
        TemplatesManager.sharedInstance.displayTemplateForSignature(templateId: templateId, controller: self, tabData: tabData, recipientData: recipientData, customFields:customFields, onlineSign: onlineSign, attachmentUrl: attachmentUrl)
        { (errMsg) in
            
            // return to client screen
            let vcIndex = self.navigationController?.viewControllers.index(where: { (viewController) -> Bool in
                if let _ = viewController as? ClientViewController {
                    return true
                }
                return false
            });
            
            let clientVC = self.navigationController?.viewControllers[vcIndex!] as! ClientViewController;
            self.navigationController?.popToViewController(clientVC, animated: true)
        }
    }

    // MARK: Private Methods
    
    private func styleUIElements()
    {
        // set custom nav title
        self.navigationItem.titleView = Bundle.main.loadNibNamed("CustomNavTitle", owner: nil, options: nil)?.first as! UIView?;
    }

}
